File: backend/app/schemas/interviews.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class Interview(BaseModel):
    id: int
    job_title: str
    company_name: str
    company_logo: Optional[str] = None
    interview_date: datetime
    duration_minutes: int
    interview_type: str
    meeting_link: str
    meeting_id: str
    meeting_password: str
    status: str
    location: str
    interviewer_name: str
    interviewer_role: str
    preparation_notes: str
    requirements: List[str]
    created_at: datetime

class InterviewResponse(BaseModel):
    id: int
    job_title: str
    company_name: str
    company_logo: Optional[str] = None
    interview_date: datetime
    duration_minutes: int
    interview_type: str
    meeting_link: str
    meeting_id: str
    meeting_password: str
    status: str
    location: str
    interviewer_name: str
    interviewer_role: str
    preparation_notes: str
    requirements: List[str]
    created_at: datetime

File: frontend/src/lib/interviews.ts
import { api } from '@/lib/api'

export interface Interview {
  id: number
  job_title: string
  company_name: string
  company_logo?: string
  interview_date: string
  duration_minutes: number
  interview_type: string
  meeting_link: string
  meeting_id: string
  meeting_password: string
  status: string
  location: string
  interviewer_name: string
  interviewer_role: string
  preparation_notes: string
  requirements: string[]
  created_at: string
}

export interface InterviewResponse {
  interviews: Interview[]
  total_interviews: number
  upcoming_interviews: number
}

export const interviewsService = {
  async getInterviews(): Promise<InterviewResponse> {
    const response = await api.get('/interviews/')
    return response.data
  }
}
